### PR TITLE
[frontend/backend] Share the result of inferrence rules in TAXII collections (#7069)

### LINKED ISSUE #7069: Share the result of inferrence rules in TAXII collections
## Use case
We have created https://github.com/OpenCTI-Platform/opencti/issues/6171 to be able to automatically add indicators/observables promoted from indicators/observables in a container.
Once this is done, we would 3 ways to achieve this: 
- use the inference rules 
- use the automation (only EE users)
- use the above-mentioned feature

The problem with the inference rule is that its results (the entities/relations created from an inference rule) are not shared within the TAXII feeds. It creates a gap in the whole use case which would be to automatically add generated indicators/observables in a container & share them in the different feeds you have set up. 

## Proposed Solution

Add an option, enabled by default, to share by default in a feed the result of inference rules.
This will not affect already existing rules.

### PR DESCRIPTION
**Pull request by ValentinBouzinFiligran**
### Proposed changes

* checkbox added to taxii creation drawer
* value created on elastic
* value usable on collectionQuery to change the index

### Related issues

* https://github.com/OpenCTI-Platform/opencti/issues/7069

### Checklist

- [ ] I consider the submitted work as finished
- [ ] I tested the code for its functionality
- [ ] I wrote test cases for the relevant uses case (coverage and e2e)
- [ ] I added/update the relevant documentation (either on github or on notion)
- [ ] Where necessary I refactored code to improve the overall quality

### CODE REVIEW

File: opencti-platform/opencti-graphql/src/modules/attributes/internalObject-registrationAttributes.ts [hunk 1]
```diff
@@ -360,6 +360,7 @@ const internalObjectsAttributes: { [k: string]: Array<AttributeDefinition> } = {
     { name: 'description', label: 'Description', type: 'string', format: 'text', mandatoryType: 'no', editDefault: false, multiple: false, upsert: true, isFilterable: true },
     { name: 'filters', label: 'Filters', type: 'string', format: 'text', mandatoryType: 'no', editDefault: false, multiple: false, upsert: false, isFilterable: false },
     { name: 'taxii_public', label: 'Public taxii', type: 'boolean', mandatoryType: 'external', editDefault: true, multiple: false, upsert: false, isFilterable: true },
+    { name: 'withInferences', label: 'With inferences', type: 'boolean', mandatoryType: 'no', editDefault: true, multiple: false, upsert: false, isFilterable: false },
```
Comment (SouadHadjiat) [inline opencti-platform/opencti-graphql/src/modules/attributes/internalObject-registrationAttributes.ts]: attributes are usually named with underscore like "with_inferences".
Maybe "include_inferences" could be more specific ?

File: opencti-platform/opencti-front/src/private/components/data/taxii/http-client.private.env.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (SouadHadjiat) [inline opencti-platform/opencti-front/src/private/components/data/taxii/http-client.private.env.json]: what is this file?

File: opencti-platform/opencti-front/src/private/components/data/taxii/TaxiiCollectionCreation.tsx [hunk 3]
```diff
@@ -88,6 +89,12 @@ const TaxiiCollectionCreation: FunctionComponent<TaxiiCollectionCreationProps> =
   const { t_i18n } = useFormatter();
   const classes = useStyles();
   const [filters, helpers] = useFiltersState(emptyFilterGroup);
+  const [areInferenceRulesResultAdded, setAreInferenceRulesResultAdded] = useState(true);
```
Comment (SouadHadjiat) [inline opencti-platform/opencti-front/src/private/components/data/taxii/TaxiiCollectionCreation.tsx]: why is `include_inferrences` in a separate state and not directly part of the form ?

File: opencti-platform/opencti-front/src/private/components/data/taxii/http-client.private.env.json (hunk 2 above)
Comment (ValentinBouzinFiligran) [reply opencti-platform/opencti-front/src/private/components/data/taxii/http-client.private.env.json]: I don't know where it comes from

File: opencti-platform/opencti-front/src/private/components/data/taxii/TaxiiCollectionCreation.tsx (hunk 3 above)
Comment (ValentinBouzinFiligran) [reply opencti-platform/opencti-front/src/private/components/data/taxii/TaxiiCollectionCreation.tsx]: updated ! :) 
thanks !

**Comment by SouadHadjiat** [comment]
When creating a taxii collection including inferrences, I can see them in the collection. Then when I create a feed ingesting this collection, I get a duplicate of my inferred relationships :

Capture d'écran 2024-06-17 171114

=> seen with product: not an issue

**Review by SouadHadjiat — APPROVED** [review]

**Pull request merged**